corehq/apps/app_manager/views/apps.py

```diff
@@ -393,26 +393,35 @@ def copy_app(request, domain):
         from corehq.apps.app_manager.views.view_generic import view_generic
         return view_generic(request, domain, app_id=app_id, copy_app_form=form)
 
+    if request.POST.get('build_id'):
+        build = Application.get(request.POST.get('build_id'))
+        build.convert_build_to_app()
+        app_id_or_source = build.export_json(dump_json=False)
+    else:
+        app_id_or_source = app_id
+
     def _inner(request, link_domain, data, master_domain=domain):
         clear_app_cache(request, link_domain)
         if data['toggles']:
             for slug in data['toggles'].split(","):
                 set_toggle(slug, link_domain, True, namespace=toggles.NAMESPACE_DOMAIN)
         linked = data.get('linked')
         if linked:
-            return _create_linked_app(request, app, link_domain, data['name'])
+            return _create_linked_app(request, app, link_domain, data['name'], master_build_id=data['build_id'])
```

Comment:
Shouldn't `app` here come from `app_id_or_source` instead?

Reply:
Good catch, this wasn't working for linked apps. Added a few commits to fix, clean up, and add a test.